### ISSUE TITLE
Support multiple values as independent fields from one-to-many relationships in reports and queries

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#2005]
This may require a special formatter as discussed in our meeting about scriplets and supporting these kinds of relationships.

> As there are many traits and different traits for different specimens within the wildlife collection, we have utilised the CollectionObjectProperties table to have a 1 to many relationship between the specimen record and the traits.  I have added a screen shot of the Traits Data part of the Collection Object form, and a sample of the BirdBoneLabel.  The question is: How do I set up the query and the iReport template to show some of the traits on the label, as the information is not a value contained in a single field?  Fish is different, as there are fewer traits, and we have a single field for each trait.

![image](https://user-images.githubusercontent.com/37256050/185667612-a12f371b-dc63-425f-8b8e-2320e9d98a97.png)

![image](https://user-images.githubusercontent.com/37256050/185667629-d8e7cef0-0aae-46e5-90e4-673b48257069.png)

> The label does have the captions, followed by the actual field value, and different records will have different information present.  The label layout needs to be consistent.  Also, with the collection object properties, different fields will be used, depending on if the value is text or numeric.  

> As another example, this is the bird skin label template for the label, and it shows how the traits are spread throughout the label, just not a small grouping of fields.

![image](https://user-images.githubusercontent.com/37256050/185667639-4f8e11b4-2dd6-429e-9598-102de63e10da.png)

**Requested By:** Rob at CSIRO

**Comment by maxxxxxdlp** [comment]
Can this be fixed by creating a formatter that formats those values into a string like `Sex: ...     Age: ...      Wgt(g): ...`?

If so, fixing https://github.com/specify/specify7/issues/1959 would allow for this use case.

Fixing https://github.com/specify/specify7/issues/23 would help with this too

**Comment by grantfitzsimmons** [comment]
@maxxxxxdlp No, it would not be fixed. Technically #1959 would provide a better workaround, but they want to display field values as separate instances apart from each other.

**Comment by maxxxxxdlp** [comment]
Can they include a tab character (`\t`) in their formatter?

**Comment by grantfitzsimmons** [comment]
@maxxxxxdlp I've never tried to do that before but that's an interesting approach. Have you established a formatter with a tab character before? Would the report runner interpret that correctly?

**Comment by maxxxxxdlp** [comment]
Never tries it, but it can work, as tab is just a character like any other. If that doesn't work, we call fallback to using 4 spaces instead (not as good of a solution as spaces don't have auto-alignment like tabs do)

**Comment by grantfitzsimmons** [comment]
Still does not resolve the issue :ok_man:. Many cases that have requested this fix do not present the options as a line like the one in the screenshot. In Specify 6, you can use scriptlets to make this happen for some relationships.

I do think it's a step in the right direction! I don't think we should use the approach 6 did due to the reasons Ben has described (Specify 6 goes back after the query has run and gathers information from the DB)

**Comment by maxxxxxdlp** [comment]
Could you replace the tab characters with new line characters to have a multiline formatter?

**Comment by grantfitzsimmons** [comment]
I'm going to have to try that. I think that's a good compromise for the time being if that works and displays properly. Would require #1959 if they have an established formatter already
